Bug: Cannot use array types in function return type declarations

## Description

The parser does not recognize array type syntax `[Type]` when used in function return type declarations after `->`. Array types work correctly in variable declarations but fail when specified as return types, causing parse errors.

## Example EZ Code

```ez
import @std
import @arrays

using std
do main() {
  println(some_thing())
}

const Person struct {
  name string
  age int
}

do some_thing() -> [Person] {
  temp p1 = new(Person)
  temp p2 = new(Person)
  temp p3 = new(Person)

  temp arr [Person] = {p1, p2, p3}  // Array declaration works fine here

  return arr
}
```

## Expected Behavior

The function should parse successfully with `[Person]` recognized as a valid return type, just like it's recognized in variable declarations. The program should:
1. Parse without errors
2. Type-check the function to ensure it returns an array of Person structs
3. Execute and print the array

## Actual Behavior

Parser fails with multiple cascading errors:

```
error[E1002]: expected return type after '->', got [ instead
  --> ./main.ez:15:18
   |
15 | do some_thing() -> [Person]{
   |                  ^^ expected token not found
   |

error[E1003]: expected expression, found ']'
  --> ./main.ez:15:27
   |
15 | do some_thing() -> [Person]{
   |                           ^ invalid syntax
   |

error[E1001]: unexpected token 'temp'
  --> ./main.ez:16:3
   |
16 |   temp p1 = new(Person)
   |   ^^^^ unexpected token here
   |

error[E1002]: expected }, got IDENT instead
  --> ./main.ez:16:8
   |
16 |   temp p1 = new(Person)
   |        ^^ expected token not found
   |

error[E1003]: expected expression, found end of block
  --> ./main.ez:23:1
   |
23 | }
   | ^ invalid syntax
   |

summary: generated 5 errors
```

The first error is the root cause - parser doesn't recognize `[` as valid start of return type. Subsequent errors are cascading failures from the initial parse error.

## Inconsistency

Array syntax `[Type]` works in these contexts:
- ✓ Variable declarations: `temp arr [Person] = {...}`
- ✓ Function parameters (likely): `do foo(arr [Person]) { ... }`
- ❌ Function return types: `do foo() -> [Person] { ... }`

This inconsistency is confusing for users.

## Implementation Notes

The issue is likely in the parser's return type parsing logic. Looking at `pkg/parser/parser.go`, the function that parses return types after `->` needs to be updated to recognize array type syntax.

The parser should:
1. Check if the next token is `[` when parsing return types
2. If so, parse the full array type syntax `[Type]` or `[Type, size]`
3. Add the parsed array type to the function's return types list

This should handle:
- Dynamic arrays: `-> [Person]`, `-> [int]`, `-> [string]`
- Fixed-size arrays: `-> [int, 10]`, `-> [Person, 5]`
- Multiple returns with arrays: `-> ([int], string, [Person])`

## Related

This may affect:
- Multiple return values with arrays: `do foo() -> ([int], string) { ... }`
- Nested array types (if supported): `do foo() -> [[int]] { ... }`
- Array of arrays: `do foo() -> [[Person]] { ... }`

## Notes

- Array types already work in variable declarations, so the type system understands them
- This is purely a parser limitation in the return type parsing context
- The fix should be relatively straightforward - extend return type parser to handle array syntax

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -1060,9 +1060,13 @@ func (p *Parser) parseFunctionDeclarationWithAttrs(attrs []*Attribute) *Function
 		if p.currentTokenMatches(LPAREN) {
 			// Multiple return types
 			stmt.ReturnTypes = p.parseReturnTypes()
-		} else if p.currentTokenMatches(IDENT) {
-			// Single return type
-			stmt.ReturnTypes = []string{p.currentToken.Literal}
+		} else if p.currentTokenMatches(IDENT) || p.currentTokenMatches(LBRACKET) {
+			// Single return type (identifier or array)
+			typeName := p.parseTypeName()
+			if typeName == "" {
+				return nil
+			}
+			stmt.ReturnTypes = []string{typeName}
 		} else {
 			// Invalid token after arrow
 			msg := fmt.Sprintf("expected return type after '->', got %s instead", p.currentToken.Type)
@@ -1223,11 +1227,30 @@ func (p *Parser) parseReturnTypes() []string {
 	p.nextToken() // move past (
 
 	for !p.currentTokenMatches(RPAREN) {
-		types = append(types, p.currentToken.Literal)
+		// Parse type name (can be IDENT or array type starting with LBRACKET)
+		if p.currentTokenMatches(IDENT) || p.currentTokenMatches(LBRACKET) {
+			typeName := p.parseTypeName()
+			if typeName == "" {
+				return nil
+			}
+			types = append(types, typeName)
+		} else {
+			msg := fmt.Sprintf("expected type name in return types, got %s", p.currentToken.Type)
+			p.addEZError(errors.E1002, msg, p.currentToken)
+			return nil
+		}
+
+		// Check for comma or end of list
 		if p.peekTokenMatches(COMMA) {
 			p.nextToken() // consume comma
+			p.nextToken() // move to next type
+		} else if p.peekTokenMatches(RPAREN) {
+			p.nextToken() // move to )
+		} else {
+			msg := fmt.Sprintf("expected ',' or ')' in return types, got %s", p.peekToken.Type)
+			p.addEZError(errors.E1002, msg, p.peekToken)
+			return nil
 		}
-		p.nextToken()
 	}
 
 	return types

